spec/cocina/models/admin_policy_shared_examples.rb

```diff
@@ -0,0 +1,246 @@
+
+  describe 'initialization' do
+    context 'with minimal required properties provided' do
+      it 'populates required attributes passed in' do
+        if required_properties[:externalIdentifier]
+          expect(admin_policy.externalIdentifier).to eq required_properties[:externalIdentifier]
+        end
+        expect(admin_policy.label).to eq required_properties[:label]
+        expect(admin_policy.type).to eq required_properties[:type]
+        expect(admin_policy.version).to eq required_properties[:version]
+      end
+
+      it 'populates non-passed required attributes with default values' do
+        expect(admin_policy.access).to be_kind_of(Cocina::Models::AdminPolicy::Access)
+        expect(admin_policy.access.attributes.size).to eq 0
+
+        expect(admin_policy.administrative).to be_kind_of(Cocina::Models::AdminPolicy::Administrative)
+        expect(admin_policy.administrative.default_object_rights).to eq admin_policy_default_rights
+        expect(admin_policy.administrative.registration_workflow).to be_nil
+        expect(admin_policy.administrative.hasAdminPolicy).to be_nil
+
+        expect(admin_policy.identification).to be_kind_of(Cocina::Models::AdminPolicy::Identification)
+        expect(admin_policy.identification.attributes.size).to eq 0
+
+        expect(admin_policy.structural).to be_kind_of(Cocina::Models::AdminPolicy::Structural)
+        expect(admin_policy.structural.attributes.size).to eq 0
+      end
+    end
+
+    context 'with a string version property' do
+      let(:properties) { required_properties.merge(version: required_properties[:version].to_s) }
+
+      it 'coerces to integer' do
+        expect(admin_policy.version).to eq required_properties[:version]
+      end
+    end
+
+    context 'with all specifiable properties' do
+      let(:properties) do
+        required_properties.merge(
+          administrative: {
+            default_object_rights: '<rightsMetadata></rightsMetadata>',
+            registration_workflow: 'wasCrawlPreassemblyWF',
+            hasAdminPolicy: 'druid:mx123cd4567'
+          },
+          description: {
+            title: [
+              {
+                primary: true,
+                titleFull: 'My admin_policy'
+              }
+            ]
+          }
+        )
+      end
+
+      it 'populates all attributes passed in' do
+        expect(admin_policy.administrative.hasAdminPolicy).to eq 'druid:mx123cd4567'
+        expect(admin_policy.administrative.default_object_rights).to eq '<rightsMetadata></rightsMetadata>'
+        expect(admin_policy.administrative.registration_workflow).to eq 'wasCrawlPreassemblyWF'
+
+        expect(admin_policy.description.title.first.attributes).to eq(primary: true,
+                                                                      titleFull: 'My admin_policy')
+      end
+
+      context 'with a string description title primary boolean property' do
```

Comment:
Why are we testing this?

Reply:
to show what happens?

Reply:
it doesn't seem to be tested elsewhere, but this may not be the right place (ditto for version attribute).

Note that File.administrative boolean property DOES coerce a string to boolean

https://github.com/sul-dlss/cocina-models/blob/master/spec/cocina/models/file_shared_examples.rb#L98-L117

But Description does NOT coerce a string to boolean.

Should we accept strings or not?  Why in some places and not others?

I can pull out this out into new description_spec, and create a version_spec if you like.

Reply:
the coercion of a version value to an integer was in the existing specs, FWIW.  Not that it means it should stay there.

Reply:
This isn't worth slowing anything down over.

Reply:
i'm gonna merge this, and work on a PR to test this stuff in a more appropriate way.